server/service/src/invoice/outbound_return/generate_outbound_return_lines.rs

```diff
@@ -0,0 +1,148 @@
+        });
+    }
+
+    let mut filter = StockLineFilter::new().is_available(true);
+
+    if !stock_line_ids.is_empty() {
+        filter.id = Some(EqualFilter::equal_any(
+            stock_line_ids.iter().map(String::clone).collect(),
+        ))
+    }
+    filter.item_id = item_id.map(|item_id| EqualFilter::equal_to(&item_id));
+
+    let stock_lines = StockLineRepository::new(&ctx.connection)
+        .query_by_filter(filter, Some(store_id.to_string()))?;
+
+    let return_lines: Vec<OutboundReturnLine> = stock_lines
+        .iter()
+        .map(|stock_line| OutboundReturnLine {
+            id: uuid(),
+            stock_line: stock_line.clone(),
+
+            reason_id: None,
+            comment: None,
+            number_of_packs: 0,
+        })
+        .collect();
+
+    // TODO:
+    // if return_id, query for return lines by return id
+
+    Ok(ListResult {
+        count: return_lines.len() as u32,
+        rows: return_lines,
+    })
+}
+
+#[cfg(test)]
+mod test {
+    use crate::service_provider::ServiceProvider;
+    use repository::{
+        mock::{mock_item_a_lines, MockDataInserts},
+        test_db::setup_all,
+    };
+
+    #[actix_rt::test]
+    async fn generate_outbound_return_lines_nothing_supplied() {
+        let (_, _, connection_manager, _) = setup_all(
+            "generate_outbound_return_lines_nothing_supplied",
+            MockDataInserts::all(),
+        )
+        .await;
+
+        let service_provider = ServiceProvider::new(connection_manager, "app_data");
+        let context = service_provider.basic_context().unwrap();
+        let service = service_provider.invoice_service;
+
+        let store_id = "store_a";
+        let stock_line_ids = vec![];
+        let item_id = None;
+        let return_id = None;
+
+        let result = service
+            .generate_outbound_return_lines(&context, store_id, stock_line_ids, item_id, return_id)
+            .unwrap();
+
+        assert_eq!(result.count, 0);
+    }
+
+    #[actix_rt::test]
+    async fn generate_outbound_return_lines_stock_line_ids() {
+        let (_, _, connection_manager, _) = setup_all(
+            "generate_outbound_return_lines_stock_line_ids",
+            MockDataInserts::all(),
+        )
+        .await;
+
+        let service_provider = ServiceProvider::new(connection_manager, "app_data");
+        let context = service_provider.basic_context().unwrap();
+        let service = service_provider.invoice_service;
+
+        let store_id = "store_a";
+        let stock_line_ids = mock_item_a_lines()
+            .iter()
+            .map(|stock_line| stock_line.id.clone())
+            .collect();
+        let item_id = None;
+        let return_id = None;
+
+        let result = service
+            .generate_outbound_return_lines(&context, store_id, stock_line_ids, item_id, return_id)
+            .unwrap();
+
+        assert_eq!(result.count, 2);
+        assert_eq!(result.rows[0].stock_line.item_row.id, "item_a");
```

Comment:
For a more complex use case (i've just added a suggestion for one in the comment, not for this PR though), we would need to sort by id i think:

https://github.com/msupply-foundation/open-msupply/blob/ae7135cb1ae2b6cd08f0ef3f92d558c04e021c90/server/service/src/item_stats.rs#L173